gnmidiff/setrequest_test.go

```diff
@@ -0,0 +1,378 @@
+// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gnmidiff
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/openconfig/ygot/exampleoc"
+	"github.com/openconfig/ygot/ygot"
+
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+)
+
+// must7951 calls Marshal7951 to create a JSON_IETF TypedValue.
+func must7951(v interface{}) *gpb.TypedValue {
+	b, err := ygot.Marshal7951(v, &ygot.RFC7951JSONConfig{AppendModuleName: true})
+	if err != nil {
+		panic(err)
+	}
+	return &gpb.TypedValue{
+		Value: &gpb.TypedValue_JsonIetfVal{
+			JsonIetfVal: b,
+		},
+	}
+}
+
+func TestMinimalSetRequestIntent(t *testing.T) {
+	tests := []struct {
+		desc         string
+		inSetRequest *gpb.SetRequest
+		wantIntent   setRequestIntent
+		wantErr      bool
+	}{{
```

Comment:
Can you add a test that has a set request with updates for something like:

"/interfaces/port"
"/interfaces/portname"

I realize these are no valid openconfig paths, but they test the issue where one leaf is a prefix of another.

Reply:
Done.